src/data/dfinsupp/interval.lean

```diff
@@ -0,0 +1,164 @@
+/-
+Copyright (c) 2021 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import data.finset.locally_finite
+import data.dfinsupp.order
+
+/-!
+# Finite intervals of finitely supported functions
+
+This file provides the `locally_finite_order` instance for `Π₀ i, α i` when `α` itself is locally
+finite and calculates the cardinality of its finite intervals.
+-/
+
+open dfinsupp finset
+open_locale big_operators pointwise
+
+variables {ι : Type*} {α : ι → Type*}
+
+namespace finset
+variables [decidable_eq ι] [Π i, has_zero (α i)] {s : finset ι} {f : Π₀ i, α i}
+  {t : Π i, finset (α i)}
+
+/-- Finitely supported product of finsets. -/
+def dfinsupp (s : finset ι) (t : Π i, finset (α i)) : finset (Π₀ i, α i) :=
```

Comment:
Should this be called something like `dbind` for dependent bind?

Reply:
I'm not sure, because it's not most general dependent bind. If you have a look at #10930, `finset.dfinsupp` is really close in spirit to `finset.finsupp`, so their names should stay close too.

Reply:
Okido